### DISCUSSION TITLE
Add example that parses valid email addressed.

### DISCUSSION THREAD

**Discussion by Luckey-Elijah** [petitparser/dart-petitparser-examples#8] [Q&A] [answered]
Email addresses can be tricky to handle with regex. I think an idiomatic example written with `petitparser` would be great  for reference.

**Comment by renggli** [comment, marked as answer]
There is a basic example here in the benchmarks (with regex comparison): https://github.com/petitparser/dart-petitparser-examples/blob/58b3b4782f78f4d49190d81bff7ffd910f796a65/bin/benchmark/regexp_benchmark.dart#L57-L63

The [URL parser](https://petitparser.github.io/examples/uri/uri.html) is probably a more complete example to extract e-mail addresses, but it requires the scheme: `mailto://foo@bar.com`. Maybe you can start [there](https://github.com/petitparser/dart-petitparser-examples/blob/58b3b4782f78f4d49190d81bff7ffd910f796a65/lib/src/uri/authority.dart)?

![email](https://user-images.githubusercontent.com/131610/235610762-4771c7e1-2add-473e-896b-5b6ad5899f4a.png)

**Comment by Luckey-Elijah** [comment]
Thank you @renggli this is a good example and what I am looking for 😄
